IPCheck/dashAnalysis.sh

```diff
@@ -1,118 +1,120 @@
+#		$3	: Name to use (usually the name of the repo) 
 #--------------------------------------------------------------
 
 MAVEN="mvn"
-DASH_LIB="libs/org.eclipse.dash.licenses-0.0.1-SNAPSHOT.jar"
-
-if [[ -z GIT_REPOS ]]; then
-	echo "Please define the variable env. GIT_REPOS with the path to the folder containing all your GIT clones!"
-	exit
-fi 
+DASH_LIB="./libs/org.eclipse.dash.licenses-0.0.1-SNAPSHOT.jar"
 
-if [[ -z $1 ]]; then
-	echo "Syntax: dashAnalysis.sh <type> <repo-name> [<destination>]"
+if [[ $# < 3 ]]; then
+	echo "ERROR: Wrong number of arguments"
+	echo "Syntax: dashAnalysis.sh <type> <source> <destination>"
 	echo '  with:   <type>         ="Maven" or "Nodes" or "Yarn"'
-	echo '          <repo-name>    = Name of the git repo or path to the folder containing the pom.xml file'
-	echo '          <destination>  = If the path of the folder is not the name of the repo then this third argument specifies the name of the repo'
+	echo '          <source>       = Path to the folder containing the config file'
+	echo '          <destination>  = Path of the folder where the results will be stored'
 	exit
 fi
 
-RES="./analysis"
-
-ECHO
-ECHO "+=====================================================+"
-ECHO "|                                                     |"
-ECHO "|                    DASH Analysis                    |"
-ECHO "|                                                     |"
-ECHO "+=====================================================+"
-ECHO
+echo
+echo "+=====================================================+"
+echo "|                                                     |"
+echo "|         Eclipse DASH License Tool Analysis          |"
+echo "|                                                     |"
+echo "+=====================================================+"
+echo
 
 # Grab arguments
-KIND_OF=$1 	# To select between "Maven" | "Nodes" | "Yarn"
-REPO=$2		# Path of the folder containing the file to parse. The repo is supposed to be located under the $GIT_REPOS folder
-DEST=$3		# If the path of the folder is not the name of the repo then this third argument can specify the name of the repo
+KIND_OF=$1 		# To select between "Maven" | "Nodes" | "Yarn"
+SRC_DIR=$2		# Path to the folder containing the config file
+DEST_DIR=$3		# Path of the folder where the results will be stored
 
-# Set source directory
-SRC_DIR=$GIT_REPOS"/"$REPO
+echo ":"
+echo ": DASH analysis of:.........."$SRC_DIR
+echo ": Type:......................"$KIND_OF
+echo ": Results are available in:.."$DEST_DIR
+echo ":"
+echo ":"
+mkdir -p $DEST_DIR 
 
-# Set Destination directory
-if [[ -n $3 ]]; then
-	DEST=$3
-else
-	DEST=$REPO
-fi
-DEST_DIR=$RES"/"$DEST
+LOG_FILE="$DEST_DIR/log.txt"
+ANALYSIS_RESULT_FILE="$DEST_DIR/03-dash_analysis.csv"
+
+# Pull changes from a remote repo
+echo ": Pull changes from a remote repo into:"$SRC_DIR
+echo ": Pull changes from a remote repo into:"$SRC_DIR > $LOG_FILE
+git pull >> $LOG_FILE
```

Comment:
I think we need to `cd` to `$SRC_DIR` before doing the pull. Right now it seems to be pulling this repo instead of the repo to be analysed.

Reply:
It should also be checked if `$SRC_DIR` exists and exit with an error message if it doesn't.

Reply:
Fixed